app/src/main/kotlin/github/com/raccok/dota2androidapp/MainActivity.kt

```diff
@@ -93,12 +104,94 @@ class MainActivity : Activity() {
     }
   }
 
+  private fun queryHeroNames() {
+    // Check if the device is connected to the internet
+    if (!deviceIsOnline()) {
+      Toast.makeText(applicationContext,
+                     "Querying heroes from the Dota 2 API failed (no internet connection)",
+                     Toast.LENGTH_LONG).show()
+      return
+    }
+
+    // Create a client to perform networking
+    val client = AsyncHttpClient()
+
+    // Additional parameters needed to query the Dota 2 API for currently available heroes
+    // (TODO: the API key is hard-coded temporarily, until we have Steam user authentication)
+    val params = "IEconDOTA2_570/GetHeroes/v1?key=F9FEEEDF9C7762EA5D76F2EEDB7A08BC&language=en_us"
```

Comment:
you'll have to rotate this key to prevent abuse.  Store it in a config file and share it with the dev team outside of github.

Reply:
Fixed, thank you!